lib/model-utils.js

```diff
@@ -391,6 +397,17 @@ ModelUtils._coerce = function(where, options) {
     if (!DataType) {
       continue;
     }
+    const value = where[p];
+    if (isNestedModel(DataType)) {
+      if (Array.isArray(DataType)) {
+        if (value === null || value === undefined) continue;
+        for (const it of value) {
+          self._coerce(it, options, DataType[0].definition);
+        }
+      } else {
+        self._coerce(value, options, DataType.definition);
+      }
+    }
```

Comment:
I might be missing something here, but by taking a quick look, if **nested** model is found, couldn't it just `continue` on to next iteration instead of following the rest of the logic?

Reply:
If the nested model is not an array, then we are recursively sending its `definition` to use with the `_coerce` function here.

Reply:
Yes I understand that. I was talking about using a `continue`  before closing the **NestedModel** is between line 409 and 410

Reply:
I see what you mean now thank you for clarifying. If the model is nested, then it either checks for array or not but either way it calls `self._coerce()`. Would it be better if I re-wrote this as follows (without the `else` block)?
```js
    if (isNestedModel(DataType)) {
      if (Array.isArray(DataType)) {
        if (value === null || value === undefined) continue;
        for (const it of value) {
          self._coerce(it, options, DataType[0].definition);
        }
      } 
        self._coerce(value, options, DataType.definition);
    }
```